pkg/wakaama/contrib/lwm2m_objects/light_control/light_control.c

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     lwm2m_objects_light_control
+ * @{
+ * @brief       Light control object implementation for LWM2M client using
```

Comment:
I was thinking maybe it is useful to have some standard objects already implemented (besides the mandatory ones). Perhaps they can be moved somewhere else? What do you think?

Reply:
> SAUL to LWM2M object module

This could make more sense to have in contrib, as it is more general. Depends on you, if you want to go through the trouble (keep in mind, how are we going to test that?)

For me, having the lights module in an example is fine, and it also serves to show how to define an application specific module, making the example richer. I personally don't like examples that are too short and "magical". Of course any application will be short if all the needed functionality besides the libraries "just happens" to be be implemented somewhere else, but I feel it would be more realistic- and thus useful for devs wanting to build a real-life application- to have the lights in the examples.

Reply:
BTW, that was the same dilemma I faced when I was writing the Lua port. At first I thought of creating another package/module with bindings between riot subsystems an lua. But after realising how easy the Lua C api was I chose to leave it up to the user to create what he needs.